### PR TITLE
[4.2] Display one set of buttons

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #39646 .

### Summary of Changes

When @laoneo did the PR for tinyMCE https://github.com/joomla/joomla-cms/pull/39625 I've contacted him with this request that was left for another PR

Screenshot 2023-01-16 at 16 53 32

### Testing Instructions

add these lines after https://github.com/joomla/joomla-cms/blob/4dcee6991fdbbef2daa20168d930f3b4e790dcb3/administrator/components/com_content/forms/article.xml#L58
```xml
<field name="articletext1" type="editor" editor="none" buttons="true" hide="menu,module" />
<field name="articletext2" type="editor" editor="codemirror" buttons="true" hide="article,image" />
```

add these lines after https://github.com/joomla/joomla-cms/blob/4dcee6991fdbbef2daa20168d930f3b4e790dcb3/administrator/components/com_content/tmpl/article/edit.php#L72

```
<?php echo $this->form->getInput('articletext1'); ?>
<?php echo $this->form->getInput('articletext2'); ?>
```

Create a new article, tinyMCE should have all the buttons, none should miss the menu and module and codemirror should miss article and image
### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Editor/Editor.php [hunk 1]
```diff
@@ -94,12 +94,10 @@ public function __construct($editor = 'none', DispatcherInterface $dispatcher =
         $this->getDispatcher()->addListener(
             'getButtons',
             function (AbstractEvent $event) {
-                $editor = $event->getArgument('editor', null);
-                $buttons = $event->getArgument('buttons', null);
-                $result = $event->getArgument('result', []);
-                $newResult = $this->getButtons($editor, $buttons);
-                $newResult = (array) $newResult;
-                $event['result'] = array_merge($result, $newResult);
+                $event['result'] = (array) $this->getButtons(
```
Comment (laoneo) [inline libraries/src/Editor/Editor.php:97]: You loose the data which is already in the result of the event

Comment (dgrammatiko) [reply libraries/src/Editor/Editor.php:97]: What do you mean? Each event is processes in series, so X editor goes through its life cycle: init, onDisplay and onDisplay it fires the event (once) `getButtons`. The events should not stack, or at least that's my understanding...

Comment (laoneo) [reply libraries/src/Editor/Editor.php:97]: In the old code, there is an array_merge of the existing result in the event with the new one.

Comment (dgrammatiko) [reply libraries/src/Editor/Editor.php:97]: The old code was wrong and that happen with the introduction of the j4 events. The code before (3) didn’t had an event, buttons like editors were just callbacks. So a callback has a defined return value and thus the concept of merging the return values is clearly wrong in this case. Check @Fedik comment https://github.com/joomla/joomla-cms/issues/39646#issuecomment-1384206618

Comment (laoneo) [reply libraries/src/Editor/Editor.php:97]: Thanks for clarification

**Comment by OctavianC** [comment]
I have tested this item :white_check_mark: successfully on 4460852a674af6d6efb4063e67a025f67d6d2e21<br /><br />Works<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39648">issues.joomla.org/tracker/joomla-cms/39648</a>.</sub>

**Comment by OctavianC** [comment]
image

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 4460852a674af6d6efb4063e67a025f67d6d2e21<br /><br />👍<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39648">issues.joomla.org/tracker/joomla-cms/39648</a>.</sub>

**Comment by laoneo** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39648">issues.joomla.org/tracker/joomla-cms/39648</a>.</sub>

**Comment by roland-d** [comment]
Thank you

**Pull request merged**